### PR TITLE
ICU-21244 Make Valgrind CI build/test check run on every Pull Request

### PR DESCRIPTION
**Pull request by erik0686**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21244
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: .ci-builds/.azure-valgrind.yml [hunk 1]
```diff
@@ -9,8 +9,7 @@
 # Only run valgrind on the main and maint branches, and
```
Comment (jefgen) [inline .ci-builds/.azure-valgrind.yml:9]: Note: I think we'd want to update this comment as well. :)

File: .ci-builds/.azure-valgrind.yml [hunk 2]
```diff
@@ -6,11 +6,9 @@
 # The Ubuntu images don't have valgrind installed by default, so we need
 # install it first.
```
Comment (jefgen) [inline .ci-builds/.azure-valgrind.yml:7]: Hm, maybe we might want to tweak the above comment to say something like this:

```
# Note: The valgrind test configuration is in a separate file
# as it used to be run independently from the regular CI builds.
```

**Comment by jefgen** [comment]
I updated the pipeline config, so if you force-push then it should kick off the new CI run. :)

**Comment by jefgen** [comment]
> I updated the pipeline config, so if you force-push then it should kick off the new CI run. :)

`CI-Valgrind` -- Nice, it's working. :)

**Review by jefgen — APPROVED** [review]
lgtm, thank you!

**Comment by erik0686** [comment]
Thank you @jefgen !!

**Pull request merged**